regular-expression.dd

```diff
@@ -198,31 +204,33 @@ $(D_S Regular Expressions,
     $(P Again the type of splitter is range, thus allowing foreach iteration.
-    Notice the usage of lookaround in regex, it's a neat trick here as
-    stripping off final punctuation is not our intention. Breaking down this example, (?<=[.?!]) part looks behind for first ., ? or !.
-    This get us half way to our goal because \s* also matches between elements of punctuation like "?!", 
+    Notice the usage of lookaround in regex, it's a neat trick here as stripping off final punctuation is
+    not our intention. Breaking down this example, (?<=[.?!]) part looks behind for first ., ? or !.
+    This get us half way to our goal because \s* also matches between elements of punctuation like "?!",
     so a negative lookahead is introduced $(I inside lookbehind) to make sure we are past all of the punctuation marks.
     Admittedly, barrage of ? and ! makes this regex rather obscure, more then it's actually is.
     Observe that there are no restrictions on contents of lookaround expressions,
-    one can go for lookahead inside lookbehind and so on. 
-    However in general it's recommended to use them sparingly, keeping them as weapon of last resort.
+    one can go for lookahead inside lookbehind and so on.
+    However in general it's recommended to use them sparingly, keeping them as the weapon of last resort.
     )
 
-    $(P Now for something completely different yet exactly the same. For one, there is an ability
-    to precompile constant regex at compile-time:
+    $(H3 Static regex)
+    $(P Let's stop going for features and think performance. And D has something to offer here.
+    For one, there is an ability to precompile constant regex at compile-time:
     )
     ---
     static r = regex("Boo-hoo");
     assert(match("What was that? Boo-hoo?", r));
     ---
     $(P Importantly it's the exact same Regex object that works through all of the API we've seen so far.
-    It's just takes less then 1 μs of run-time to initialize.
-    Run-time version took around 10-20 μs on my machine, 
-    keep in mind that it was a trivial pattern.
-    )
-    $(P Now stepping futher in this direction there is an ability to construct specialized
-    native machine code that matches a given regex to speed up matching.
-    All of this at compile time, and again the module usage stays consistent!
+    It's just takes next to nothing to initialize, just copy over ready-made structure from the data segment.
+
+    Roughly ~ 1 μs of run-time to initialize. Run-time version took around 10-20 μs on my machine, keep in mind that it was a trivial pattern.
+
+    $(P Now stepping further in this direction there is an ability to construct specialized
+    D code code that matches a given regex and compile it instead of using default run-time engine.
+    Isn't it so often the case that code starts with regular expressions only to be later re-written
+    to heaps of scary-looking code to match speed requirements. No need to rewrite - we got you covered.
```

Comment:
> Isn't it so often the case that code starts with regular expressions only to be later re-written to heaps of scary-looking code to match speed requirements.

Either end this with a question mark, or reword the beginning to be a statement.